core/trino-spi/src/main/java/io/trino/spi/connector/ConstraintApplicationResult.java

```diff
@@ -35,7 +35,7 @@ public class ConstraintApplicationResult<T>
      */
     public ConstraintApplicationResult(T handle, TupleDomain<ColumnHandle> remainingFilter, ConnectorExpression remainingExpression, boolean precalculateStatistics)
```

Comment:
This constructor expects `remainingExpression` to be not null. We should add another constructor without the expression (or expose the Optional form.

Reply:
The constructor was removed that seems to be used by 3rd party connectors

Reply:
https://github.com/trinodb/trino/pull/23272

Reply:
Seems like an Optional.empty remainingExpression is a valid thing by looking at the `PushPredicateIntoTableScan`

Reply:
if it's not, it shouldn't be an Optional anymore

Reply:
Actually, `remainingExpression` should not even be optional -- the caller should use `TRUE` when there's no remaining predicate.